internal/rpc/rpcserver/server.go

```diff
@@ -2015,16 +2039,14 @@ func (s *walletServer) ValidateAddress(ctx context.Context, req *pb.ValidateAddr
 		}
 		result.PubKeyAddr = pubKeyAddr.String()
 	case wallet.P2SHAddress:
-		result.IsScript = true
-		_, script := ka.PaymentScript()
-		result.PayToAddrScript = script
+		_, redeem := ka.RedeemScript()
```

Comment:
I'm glad you're fixing this.  You'll see my comments when you rebase:
https://github.com/decred/dcrwallet/commit/dbdf7aa90ac2c83dc050db2b371cd02f54e9c893#diff-a23056f9a8432e77f80e98bb56a4207adbd2412877c66818836b2256602beb39R2031
This was one of a few things wrong with this method.